enhancements/microshift/kubernetes-for-industrial-edge.md

```diff
@@ -0,0 +1,436 @@
+upstream of MicroShift. There is community support for MicroShift and
+other similar projects, such as [k3s](https://k3s.io), which should
+help us if we encounter problems with the technical direction of those
+dependencies.
+
+### Drawbacks
+
+
+## Design Details
+
+### Open Questions [optional]
+
+1. How can we best minimize the window of time when OCP components are
+   not using the same version of kubernetes?
+2. Can we further reduce the dependencies of MicroShift by making
+   changes in any of the upstream components?
+3. Given the embedded image references, how will we build an
+   OKD-equivalent version of MicroShift?
+
+### Test Plan
+
+We will run the relevant kubernetes and OpenShift compliance tests in
+CI.
+
+We will have end-to-end CI tests for many deployment scenarios.
+
+We will have manual and automated QE for other scenarios.
+
+### Graduation Criteria
+
+#### Dev Preview -> Tech Preview
+
+- Ability to build MicroShift images and deploy them
+- Sufficient test coverage
+- Gather feedback from users rather than just developers
+
+#### Tech Preview -> GA
+
+The first release will have "limited availability" to a few partners
+and customers.
+
+- CSI driver selected
+- CNI driver selected
+- Reliable automated rebase process in place
+- More testing
+- User facing documentation created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+- Relative API stability
+
+#### Full GA
+
+- Available by default
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+See the discussion of upgrades earlier.
+
+### Version Skew Strategy
+
+There is no version skew for most MicroShift components because they
+are built in or running from images for which the references are built
+in. We expect to use the same CRI-O version distributed with each
+version of OCP, although that will not be rebuilt for MicroShift.
+
+### Operational Aspects of API Extensions
+
+No API extensions.
+
+#### Failure Modes
+
+N/A
+
+#### Support Procedures
+
+N/A
+
+## Implementation History
+
+* [openshift/microshift](https://github.com/openshift/microshift)
+* [Design guidelines](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+## Alternatives
+
+### RHEL
+
+Using non-containerized deployment options on RHEL may be appropriate
+for some cases, but does not solve the case where a customer is using
+multiple variations of OpenShift and wants to construct their
+application deployment independently of the cluster footprint.
+
+### podman
+
+Podman is another tool for deploying container-based workloads onto
+RHEL systems. It does not support the kubernetes and OpenShift APIs
+that users will be used to seeing in single-node and full OpenShift
+clusters, and users have asked us to provide a way to have a
+consistent deployment experience for all topologies and sizes.
+
+### Single-node OpenShift
+
+Single-node OpenShift uses far more resources than are available in
```

Comment:
Apart from resources, SNO is installed (not imaged), requires out-of-band management, implements a roll-forward (not rollback) model, targets server hardware, is not designed for users building own system images.

Reply:
added